AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -197,99 +197,96 @@ - (void)enqueueItem:(id<MSLog>)item flags:(MSFlags)flags {
+                return;
+              }
+              BOOL succeeded = response.statusCode == MSHTTPCodesNo200OK;
 
               // Success.
-              if (response.statusCode == MSHTTPCodesNo200OK) {
+              if (succeeded) {
                 MSLogDebug([MSAppCenter logTag], @"Log(s) sent with success, batch Id:%@.", ingestionBatchId);
 
                 // Notify delegates.
                 [self enumerateDelegatesForSelector:@selector(channel:didSucceedSendingLog:)
                                           withBlock:^(id<MSChannelDelegate> delegate) {
-                                            for (id<MSLog> aLog in logArray) {
+                                            for (id<MSLog> aLog in container.logs) {
                                               [delegate channel:self didSucceedSendingLog:aLog];
                                             }
                                           }];
+              }
 
-                // Remove from pending logs and storage.
-                [self.pendingBatchIds removeObject:ingestionBatchId];
-                [self.storage deleteLogsWithBatchId:ingestionBatchId groupId:self.configuration.groupId];
-
-                // Try to flush again if batch queue is not full anymore.
-                if (self.pendingBatchQueueFull && self.pendingBatchIds.count < self.configuration.pendingBatchesLimit) {
-                  self.pendingBatchQueueFull = NO;
-                  if (self.availableBatchFromStorage) {
-                    [self sendLogArray:logArray withBatchId:batchId andAuthToken:tokenInfo];
-                  }
-                }
-              } else {
+              // Failure.
+              else {
                 MSLogError([MSAppCenter logTag], @"Log(s) sent with failure, batch Id:%@, status code:%tu", ingestionBatchId,
                            response.statusCode);
 
                 // Notify delegates.
                 [self enumerateDelegatesForSelector:@selector(channel:didFailSendingLog:withError:)
                                           withBlock:^(id<MSChannelDelegate> delegate) {
-                                            for (id<MSLog> aLog in logArray) {
+                                            for (id<MSLog> aLog in container.logs) {
                                               [delegate channel:self didFailSendingLog:aLog withError:error];
                                             }
                                           }];
+              }
 
-                // Remove from pending logs.
-                [self.pendingBatchIds removeObject:ingestionBatchId];
-                [self.storage deleteLogsWithBatchId:ingestionBatchId groupId:self.configuration.groupId];
+              // Remove from pending batches and storage.
+              [self.pendingBatchIds removeObject:ingestionBatchId];
+              [self.storage deleteLogsWithBatchId:ingestionBatchId groupId:self.configuration.groupId];
```

Comment:
Double check this line, please:
1) We try to send logs
2) Success\Fail check
3) No matter the result - we delete logs from storage.

Probs need to check the result for this line

Reply:
This hasn't been changed in the PR:
https://github.com/Microsoft/AppCenter-SDK-Apple/blob/95b1d74f3bb7b22f6fa23323ac8bff230f59e9b5/AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m#L274-L276

Reply:
Resolved

Reply:
After a closer look, it looks like a real bug even on the `master` branch